SPIRV: Set NoPHIs property after rewriting them

arsenm opened the pull request:
There should be no PHIs after selection, as OpPhi is used
 instead. This hopefully avoids errors in #135277.

arsenm committed:
SPIRV: Set NoPHIs property after selection

There should be no PHIs after selection, as OpPhi is used
instead. This hopefully avoids errors in #135277.

arsenm commented:
* **#136327** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/136327?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/136327?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

mikhailramalho commented:
The errors mentioned in PR #135277 still happen if we apply the patch here without 129347733db518861cd36aa0a293334f4185ddcb:
```
Failed Tests (15):
  LLVM :: CodeGen/AArch64/fast-isel-cmp-vec.ll
  LLVM :: CodeGen/AArch64/wineh4.mir
  LLVM :: CodeGen/AArch64/wineh8.mir
  LLVM :: CodeGen/SPIRV/assume.ll
  LLVM :: CodeGen/SPIRV/expect.ll
  LLVM :: CodeGen/SPIRV/hlsl-resources/BufferStore.ll
  LLVM :: CodeGen/SPIRV/opt-gepoperator-of-gvar.ll
  LLVM :: CodeGen/SPIRV/optimizations/add-check-overflow.ll
  LLVM :: CodeGen/SPIRV/optimizations/switch-condition-type.ll
  LLVM :: Frontend/HLSL/empty_cs_entry.ll
  LLVM-Unit :: Target/SPIRV/./SPIRVTests/40/56
  LLVM-Unit :: Target/SPIRV/./SPIRVTests/41/56
  LLVM-Unit :: Target/SPIRV/./SPIRVTests/42/56
  LLVM-Unit :: Target/SPIRV/./SPIRVTests/43/56
  LLVM-Unit :: Target/SPIRV/./SPIRVTests/49/56
```

VyacheslavLevytskyy commented:
@arsenm  We currently do not select OpPhi during the instruction selection pass, instead we keep PHI until the module layout finalization stage (introduced as one of improvements in https://github.com/llvm/llvm-project/pull/119202 -- https://github.com/VyacheslavLevytskyy/llvm-project/blob/3e74d2c6983af986b70060ca44e6e1fbcb34a554/llvm/lib/Target/SPIRV/SPIRVInstructionSelector.cpp#L3354-L3369).

arsenm committed:
SPIRV: Set NoPHIs property after rewriting them

There should be no PHIs after selection, as OpPhi is used
instead. This hopefully avoids errors in #135277.

arsenm committed:
Revert "SPIRV: Set NoPHIs property after selection"

This reverts commit 552988051b0b15ca3bb0c6b98f61af46db852c7f.
